libethereum/Block.cpp

```diff
@@ -692,6 +693,29 @@ void Block::performIrregularModifications()
 	}
 }
 
+void Block::updateBlockhashContract()
+{
+	u256 const blockNumber = info().number();
+
+	u256 const metropolisForkBlock = m_sealEngine->chainParams().u256Param("metropolisForkBlock");
+	if (blockNumber == metropolisForkBlock)
+	{
+		m_state.createContract(c_blockhashContractAddress);
+		m_state.setNewCode(c_blockhashContractAddress, bytes(c_blockhashContractCode));
+		m_state.commit(State::CommitBehaviour::KeepEmptyAccounts);
+	}
+
+	if (blockNumber >= metropolisForkBlock)
+	{
+		u256 const nonce = transactionsFrom(SystemAddress);
+		u256 const gas = 1000000;
+		Transaction tx(0, 0, gas, c_blockhashContractAddress, m_previousBlock.hash().asBytes(), nonce);
```

Comment:
`Transaction` constructor takes `bytes`, not `bytes_ref`. Do you want to change that?